File: mobile/src/types/adminClaimReview.ts
export type AdminClaimStatus =
  | "all"
  | "pending"
  | "approved"
  | "rejected";

export type AdminClaimSort =
  | "newest"
  | "oldest"
  | "highest_amount"
  | "lowest_amount"
  | "longest_distance"
  | "therapist_name";

export interface AdminClaimReviewFilters {
  status: AdminClaimStatus;
  therapistId: number | null;
  therapistName: string | null;
  fromDate: string | null;
  toDate: string | null;
  minimumAmount: number | null;
  maximumAmount: number | null;
  minimumDistance: number | null;
  maximumDistance: number | null;
  search: string;
  sort: AdminClaimSort;
}

export interface AdminClaimReviewSummary {
  pendingClaims: number;
  todaysClaims: number;
  pendingAmount: number;
  highValueClaims: number;
  averageClaimAmount: number;
  averageDistance: number;
}

export interface AdminClaimReviewItem {
  id: number;
  therapistId: number;
  therapistName: string;
  therapistRole: string;
  claimDate: string;
  submittedAt: string | null;
  status: string;
  patientName: string | null;
  patientCount: number;
  visitedCount: number;
  travelDate: string | null;
  fromAddress: string | null;
  toAddress: string | null;
  totalKm: number;
  perKmRate: number;
  travelTotal: number;
  dailyAllowance: number;
  grandTotal: number;
  notes: string | null;
  isHighValue: boolean;
  isUrgent: boolean;
  ageDays: number;
}

export interface AdminClaimReviewPage {
  items: AdminClaimReviewItem[];
  page: number;
  pageSize: number;
  total: number;
  totalPages: number;
  highValueThreshold: number;
  summary: AdminClaimReviewSummary;
}

File: backend/app/schemas/admin_claim_review.py
from datetime import date, datetime

from pydantic import BaseModel


class AdminClaimReviewSummary(BaseModel):
    pending_claims: int
    todays_claims: int
    pending_amount: float
    high_value_claims: int
    average_claim_amount: float
    average_distance: float


class AdminClaimReviewItem(BaseModel):
    id: int
    therapist_id: int
    therapist_name: str
    therapist_role: str
    claim_date: date
    submitted_at: datetime | None
    status: str
    patient_name: str | None
    patient_count: int
    visited_count: int
    travel_date: datetime | None
    from_address: str | None
    to_address: str | None
    total_km: float
    per_km_rate: float
    travel_total: float
    daily_allowance: float
    grand_total: float
    notes: str | None
    is_high_value: bool
    is_urgent: bool
    age_days: int


class AdminClaimReviewResponse(BaseModel):
    items: list[AdminClaimReviewItem]
    page: int
    page_size: int
    total: int
    total_pages: int
    high_value_threshold: float
    summary: AdminClaimReviewSummary
